marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -142,7 +146,20 @@ protected JsonNode determineIngestionEndpointParams(Map<String, String> options)
         }
 
         if (doesNotHaveApiPath) {
-            endpointParams.put("apiPath", "/data-hub/5/data-services/ingestion/bulkIngester.api");
+            String apiPath = "/data-hub/5/data-services/ingestion/bulkIngester.api";
```

Comment:
Pinging @SameeraPriyathamTadikonda and @ehennum  about this - I believe we should move the initialization to HubDataSourceWriter. This is based on my assumption that @SameeraPriyathamTadikonda  will be adding the initialization and finalization logic there for the job document. 

I think it makes sense though to not write this module if Ernie specifies his own ingestion path. Thus, it seems that all of the logic in this constructor should move into HubDataWriteSource. Does that seem right?